create_infotexts.py

```diff
@@ -333,5 +230,145 @@ def main():
+            self.data["depicted_people"] = names_string
+            return
+
+        wikitext_names = []
+        for name in names:
+            wikitext_names.append(self.select_best_mapping_for_depicted_person(name))
+
+        self.data["depicted_people"] = "/".join(wikitext_names)
+
+
+
+    @staticmethod
+    def isolate_name(names_string):
+        """
+        Try isolating names and flipping names.
+
+        Expected input is a string with 'last1, first1, last2, first2...'
+
+        :param names_string: string representing depicted people
+        :return: list of flipped names.
+        :raises: ValueError if uneven number of names
+        """
+        if not names_string:
+            return []
+
+        if len(names_string.split(", ")) % 2 != 0:
+            raise ValueError("Uneven number of names.")
+
+        # Pairwise group name parts
+        words = names_string.split(", ")
+        span = 2
+        grouped_names = [", ".join(words[i:i + span]) for i in range(0, len(words), span)]
+
+        names = []
+        for name_pair in grouped_names:
+            flipped_name = helpers.flip_name(name_pair)
+            names.append(flipped_name)
+        return names
+
+    def select_best_mapping_for_depicted_person(self, flipped_name):
+        """
+        Lookup available mappings for a string respresenting a name and return best choice.
+        
+        people_mapping is complete by design.
+        
+        The priority order is wikidata item < commons category < the name only.
+        :flipped_name: string representing full name turned around from
+            e.g. "surname, given name" -> "given name surname"
+            note that give_name is normally one word e.g. "Erfraim"
+        :return:  string representing the selcted mapping value
+        """
+        name_map = people_mapping[flipped_name]
+        name_as_wikitext = ""
+        if "wikidata" in name_map.keys():
+            name_as_wikitext = "[[:d:{wd_item}|{name}]]".format(
+                wd_item=name_map["wikidata"],
+                name=name_map["name"])
+
+        elif "commonscat" in name_map.keys():
+            name_as_wikitext = "[[Category:{commonscat}|{name}]]".format(
+                commonscat=name_map["commonscat"],
+                name=name_map["name"])
+
+        else:
+            name_as_wikitext = name_map["name"]
+
+        if "commonscat" in name_map.keys():
+            self.content_cats.append(name_map["commonscat"])
+
+        return name_as_wikitext
+
+
+    def process_depicted_place(self, place_string, item):
+        """
+        Create wikiformat depicted place string from raw input data.
+        
+        Populates data["depicted_place"] and interacts with data["commonscat"] and data["meta_cats"].
+        
+        :param place_string: string value <Ort, foto> in metadata item.  
+        :return: None (output stored in object attribute
+        """
+        places_mapping = load_places_mapping()
```

Comment:
Done - moved and altered data flow for places_mapping, thus avoiding loading for each item.